### ISSUE TITLE
Function calls are not supported in decorators but 'ReduxDevtoolsPluginModule' was called.

### ISSUE THREAD

**Issue by xmlking** [amcdnl/ngxs#71]
trying to build with  `ng build --app=default --prod -oh=media  -e mock` and getting above error. 
project I am testing: https://github.com/xmlking/nx-starter-kit

```
ERROR in Error during template compile of 'CoreModule'
  Function calls are not supported in decorators but 'ReduxDevtoolsPluginModule' was called.
```

**Comment by amcdnl** [comment]
This is really odd. Our demo runs AoT w/ that plugin and has no issues.

**Comment by xmlking** [comment]
Hi @amcdnl you can reproduce it by comment/uncomment this [section] (https://github.com/xmlking/nx-starter-kit/blob/master/apps/default/src/app/core/core.module.ts#L19) and try to do prod build. `ng build --prod -oh=media`

**Comment by un33k** [comment]
I am getting this too, on all plugins.

**Comment by deebloo** [comment]
ill take a look as soon as I can

**Comment by deebloo** [comment]
@xmlking @un33k  i am having trouble reproducing. our integration test uses aot by default. are we sure this is happening in the next branch?

**Comment by xmlking** [comment]
@deenloo I am using latest released version.  What kind of trouble you see to reproduce ? If you checkout above repo and run `ng build --app=default --prod -oh=media -e mock` , you will see the error in red color

**Comment by deebloo** [comment]
I can reproduce the error in your repo but not in ours. Can your reproduce
in a simpler example like a fresh CLI project?

**Comment by xmlking** [comment]
Sure , I do as soon I reach home . Are you using angular-cli ?

**Comment by un33k** [comment]
I have looked at your integration app. It is bare-bone. Perhaps if you add full mutations/action/event the problem might happen. Also the integration app consumes the ts files directly as per your tsconfig.js We on the other hand consume the js file. Perhaps some anomaly may have been introduced during competition of Ngxs. 

I am getting other exceptions as well. Didn’t raise issues as I saw drastic changes on the next branch. 

With that said we may want to wait for the next branch to be merged in and it might fix the issues. 

Really like the concept and hope it becomes more stable prior to the Angular Air show & tell next week.

**Comment by deebloo** [comment]
I want to add more to the integration tests. And good point about how it is consumed. I'll update it to use the compiled lib

**Comment by un33k** [comment]
@deebloo I pulled the ngxs directory into my project as a lib to test. The problem went away. So very highly likely, the packaging process is causing the anomaly.

**Comment by deebloo** [comment]
Awesome find. I am working on updating integration today. Thanks!

**Comment by vdmelo** [comment]
Hello guys, thanks for the awesome job with this library.

I am having this error too. 
Link: [Repository with this error](https://github.com/vdmelo/ngxs-build-error.git)

**Execute:**
`npm run build`

**And you will see:**
> ERROR in Error during template compile of 'AppModule'
  Function calls are not supported in decorators but 'LoggerPluginModule' was called.

**Comment by deebloo** [comment]
@amcdnl definitely seems to be a build issue. I might have to move quite a few things around to fix some of it

**Comment by amcdnl** [comment]
We use ng-packgr , maybe its a config issue?

**Comment by deebloo** [comment]
I think it is how we have the project setup. I am looking at something now that may help

**Comment by deebloo** [comment]
good news and bad new folks. good news I can reproduce. bad news idk why yet :)

**Comment by deebloo** [comment]
should be fixed in 2.0. closing
